modules/event-objects/src/main/java/com/intuit/wasabi/eventobjects/EventEnvelopePayload.java

```diff
@@ -105,7 +105,7 @@ public String toJson() {
                 event != null && event.getTimestamp() != null ? event.getTimestamp().getTime(): null);
         eventJson.put("eventType", event != null ? event.getType() + "": "null");
         eventJson.put("eventName", event != null ? event.getName() + "": "null");
-        eventJson.put("eventPayload", event != null ? event.getPayload(): null);
+        eventJson.put("eventPayload", event != null && event.getPayload() != null ? event.getPayload().toString(): null);
```

Comment:
@abhishekjain88 Mind adding a quick comment to indicate why we need to do ".toString()"?